homeassistant/components/wemo/light.py

```diff
@@ -150,45 +152,59 @@ def available(self):
-            self.wemo.set_color(xy_color, transition=transitiontime)
+            if hs_color is not None:
+                xy_color = color_util.color_hs_to_xy(*hs_color)
+                self.wemo.set_color(xy_color, transition=transitiontime)
 
-        if ATTR_COLOR_TEMP in kwargs:
-            colortemp = kwargs[ATTR_COLOR_TEMP]
-            self.wemo.set_temperature(mireds=colortemp, transition=transitiontime)
+            if ATTR_COLOR_TEMP in kwargs:
+                colortemp = kwargs[ATTR_COLOR_TEMP]
+                self.wemo.set_temperature(mireds=colortemp, transition=transitiontime)
 
-        if ATTR_BRIGHTNESS in kwargs:
-            brightness = kwargs.get(ATTR_BRIGHTNESS, self.brightness or 255)
-            self.wemo.turn_on(level=brightness, transition=transitiontime)
-        else:
-            self.wemo.turn_on(transition=transitiontime)
+            if ATTR_BRIGHTNESS in kwargs:
+                brightness = kwargs.get(ATTR_BRIGHTNESS, self.brightness or 255)
+                self.wemo.turn_on(level=brightness, transition=transitiontime)
+            else:
+                self.wemo.turn_on(transition=transitiontime)
+        except ActionException as err:
+            _LOGGER.warning("Error while turning on device %s (%s)", self.name, err)
+            self._available = False
 
     def turn_off(self, **kwargs):
         """Turn the light off."""
         transitiontime = int(kwargs.get(ATTR_TRANSITION, 0))
-        self.wemo.turn_off(transition=transitiontime)
+
+        try:
+            self.wemo.turn_off(transition=transitiontime)
+        except ActionException as err:
+            _LOGGER.warning("Error while turning off device %s (%s)", self.name, err)
+            self._available = False
 
     def _update(self, force_update=True):
         """Synchronize state with bridge."""
-        self._update_lights(no_throttle=force_update)
-        self._state = self.wemo.state
+        try:
+            self._update_lights(no_throttle=force_update)
+            self._state = self.wemo.state
 
-        self._is_on = self._state.get("onoff") != 0
-        self._brightness = self._state.get("level", 255)
-        self._color_temp = self._state.get("temperature_mireds")
-        self._available = True
+            self._is_on = self._state.get("onoff") != 0
+            self._brightness = self._state.get("level", 255)
+            self._color_temp = self._state.get("temperature_mireds")
+            self._available = True
 
-        xy_color = self._state.get("color_xy")
+            xy_color = self._state.get("color_xy")
 
-        if xy_color:
-            self._hs_color = color_util.color_xy_to_hs(*xy_color)
-        else:
-            self._hs_color = None
+            if xy_color:
+                self._hs_color = color_util.color_xy_to_hs(*xy_color)
+            else:
+                self._hs_color = None
+        except (AttributeError, ActionException) as err:
```

Comment:
Please use try... except... else, so we only wrap the line that can raise inside try... except.